### PR TITLE
chore(component-testing): CT-142 update socket-ct

### PR DESCRIPTION
**Pull request by lmiller1990**

### CODE REVIEW

File: packages/server-ct/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare namespace SocketIO {
+  interface Socket {
+    inReporterRoom: boolean
+    inRunnerRoom: boolean
+  }
+}
```
Comment (JessicaSachs) [inline packages/server-ct/index.d.ts:6]: Newline?

File: packages/server-ct/src/socket-ct.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-const Bluebird = require('bluebird')
-const socketIo = require('@packages/socket')
-const editors = require('@packages/server/lib/util/editors')
-const { openFile } = require('@packages/server/lib/util/file-opener')
-const open = require('@packages/server/lib/util/open')
-const errors = require('@packages/server/lib/errors')
-const specsUtil = require('@packages/server/lib/util/specs')
+import _ from 'lodash'
+import _debug from 'debug'
+const debug = _debug('cypress:server-ct:socket')
```
Comment (JessicaSachs) [inline packages/server-ct/src/socket-ct.ts:5]: Move this below the imports (I'm surprised that there's no linter error for it)

File: packages/server-ct/src/socket-ct.ts [hunk 3]
```diff
@@ -33,26 +34,30 @@ const reporterEvents = [
...
     this.onTestFileChange = this.onTestFileChange.bind(this)
   }
 
-  onTestFileChange (filePath) {
+  onTestFileChange (filePath: string) {
     debug('test file changed %o', filePath)
   }
 
-  toReporter (event, data) {
+  toReporter (event: string, data: any) {
```
Comment (JessicaSachs) [inline packages/server-ct/src/socket-ct.ts:56]: Would it make sense to define a set of enums for the valid events?

**Review by JessicaSachs — CHANGES_REQUESTED** [review]
Probably wanna do the newline change. Everything else is just questions/optional

File: packages/server-ct/src/socket-ct.ts (hunk 3 above)
Comment (lmiller1990) [reply packages/server-ct/src/socket-ct.ts:56]: sure would, is there an easy way to get a list of all valid events for this one?

Comment (JessicaSachs) [reply packages/server-ct/src/socket-ct.ts:56]: I think here:

https://github.com/cypress-io/cypress/pull/9625/files#diff-0ea726fd88825aefe4de2a52bbdddccfe1bca5436607f3b29fd93bf5840e05abR14-L35

File: packages/server-ct/src/socket-ct.ts [hunk 4]
```diff
@@ -22,7 +35,16 @@ const runnerEvents = [
   'run:end',
 ]
 
-const reporterEvents = [
+type ReporterEvent =
```
Comment (elevatebart) [inline packages/server-ct/src/socket-ct.ts:38]: shoudl we export this type?

**Review by elevatebart — APPROVED** [review]
one little question

File: packages/server-ct/src/socket-ct.ts (hunk 4 above)
Comment (lmiller1990) [reply packages/server-ct/src/socket-ct.ts:38]: let's do it when we need it

**Pull request merged**